docstr_coverage/coverage.py

```diff
@@ -63,6 +60,71 @@ def do_ignore_node(filename: str, base_name: str, node_name: str, ignore_names:
     return False
 
 
+def analyze_docstrings(
+    file_result: File,
+    base: str,
+    node: Tuple[str, bool, List],
+    filename,
+    ignore_config: IgnoreConfig,
+):
+    """Track the existence of a docstring for `node`, and accumulate stats regarding
+    expected and encountered docstrings for `node` and its children (if any).
+
+    Parameters
+    ----------
+    file_result: File
+        The result-collection file instance on which the docstring presence is recorded.
```

Comment:
The wording of the `file_result` parameter is a bit confusing. After understanding what's going on, it made me think that the `analyze_docstrings` function would be better suited as a class method, whose properties it modifies. It seems like a reversal of the normal way of doing this sort of operation to be passing in the instance to a function, so it can call a method on that `file_result` instance to update the `file_result`'s data. Having said that, I don't really have any suggestions at the moment. I'm just a bit concerned about the confusing (at least to me) order here. I'd love to hear your thoughts on this

Reply:
I completely agree. `result_collection.File` is definitely NOT the place to put this.
You make a good point that we could just make this function private, and rename `file_result` to emphasize its mutability, but it still feels a bit strange to me. 
What do you think about making `analyze_docstrings` a (possibly renamed) method on `result_collection.ResultCollection`? I think this fits more cleanly into the `coverage.analyze` being the single point of contact, which both instantiates `ResultCollection` and calls the new method to build up the result collection. 
What are your thoughts?

Reply:
I do not like the idea of having ast logic anywhere in the `results_collection` module. The `results_collection` module is a very small in-memory data storage: it only stores data and provides utilities to iterate over and aggregate the stored data. It should not contain any logic. As such, the actual docstr-coverage logic does not belong into the data-classes, but into the logic modules (where they are now). Note that this function has always been in the current module another functions inner function; with this PR I merely fulfil the TODO to make it a ‘standalone’ function. If you prefer it as an inner function (thus avoiding some confusing parameters), that might be a good solution after all. Should I refactor back in?

Also, this explanation made me think if we should use a ‘real’ in-memory db (e.g. sqlite) instead of the `result_collection`. I see pros and cons, but they are more related to the interface we want to expose in the long term. Thus, let us take this offline as well 😊

Reply:
You make a lot of excellent points. As we discussed, let's move forward with giving the `analyze_docstrings` functions an underscore prefix, and adding some more descriptive documentation to the `file_result` parameter to make its role clearer. Again, thank you for taking the time to discuss the best solution!